### PR TITLE
Made FlutterTaskQueue visible for Swift.

### PR DESCRIPTION
**Pull request by gaaclarke**
fixes https://github.com/flutter/flutter/issues/118832

This makes the `FlutterTaskQueue` protocol visible, but keeps the `dispatch:` selector private.  Under the covers there are runtime checks to make sure `dispatch:` is available since we can't have compile time checks anymore.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by gaaclarke** [comment]
We don't have infrastructure to test this currently.  I added an issue for it: https://github.com/flutter/flutter/issues/137368

File: shell/platform/darwin/ios/platform_message_handler_ios.mm [hunk 1]
```diff
@@ -118,12 +118,16 @@ - (void)dispatch:(dispatch_block_t)block {
                                                   FlutterBinaryMessageHandler handler,
                                                   NSObject<FlutterTaskQueue>* task_queue) {
   FML_CHECK(platform_task_runner_->RunsTasksOnCurrentThread());
+  // Use `respondsToSelector` instead of `conformsToProtocol` to accomodate
+  // hacking in your own `FlutterTaskQueue`.
```
Comment (hellohuanlin) [inline shell/platform/darwin/ios/platform_message_handler_ios.mm:122]: can you explain more about "hacking in your own `FlutterTaskQueue`?

File: shell/platform/darwin/ios/flutter_task_queue_dispatch.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FLUTTER_TASK_QUEUE_DISPATCH_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_IOS_FLUTTER_TASK_QUEUE_DISPATCH_H_
+
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterBinaryMessenger.h"
+
+@protocol FlutterTaskQueueDispatch <FlutterTaskQueue>
```
Comment (hellohuanlin) [inline shell/platform/darwin/ios/flutter_task_queue_dispatch.h:10]: can you add some comments on why we need `FlutterTaskQueueDispatch`?

File: shell/platform/darwin/ios/platform_message_handler_ios.mm (hunk 1 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/platform_message_handler_ios.mm]: done, `injecting` is a better description

File: shell/platform/darwin/ios/flutter_task_queue_dispatch.h (hunk 2 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/flutter_task_queue_dispatch.h:16]: Yep, done.

**Review by hellohuanlin — APPROVED** [review]
LGTM!

**Pull request merged**